### PR TITLE
audio-mixer optimizations

### PR DESCRIPTION
**Pull request by birarda**
This PR contains the bulk of the improvements used for the improved audio-mixer for the September 7 load test. It likely still has the crash that was happening during that event as we do not yet have a fix.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/13152/HighFidelity-Beta-PR14047-dbea104.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13152/HighFidelity-Beta-PR14047-dbea104.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13152/HighFidelity-Beta-Interface-PR14047-dbea104.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13152/HighFidelity-Beta-PR14047-dbea104.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13152/HighFidelity-Beta-Interface-PR14047-dbea104.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/13200/HighFidelity-Beta-PR14047-df18b35.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13200/HighFidelity-Beta-PR14047-df18b35.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13200/HighFidelity-Beta-Interface-PR14047-df18b35.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13200/HighFidelity-Beta-PR14047-df18b35.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13200/HighFidelity-Beta-Interface-PR14047-df18b35.exe

File: assignment-client/src/audio/AudioMixer.cpp [hunk 1]
```diff
@@ -455,26 +452,26 @@ void AudioMixer::start() {
...
-    auto nextTimestamp = timestamp + std::chrono::microseconds(AudioConstants::NETWORK_FRAME_USECS);
     auto now = p_high_resolution_clock::now();
 
     // compute how long the last frame took
-    auto duration = std::chrono::duration_cast<std::chrono::microseconds>(now - timestamp);
+    auto duration = chrono::duration_cast<chrono::microseconds>(now - _startFrameTimestamp);
 
-    // set the new frame timestamp
-    timestamp = std::max(now, nextTimestamp);
+    _idealFrameTimestamp += chrono::microseconds(AudioConstants::NETWORK_FRAME_USECS);
```
Comment (Atlante45) [inline assignment-client/src/audio/AudioMixer.cpp]: I still think we need to add some frame skip recovery here.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/13265/HighFidelity-Beta-PR14047-471cda8.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13265/HighFidelity-Beta-PR14047-471cda8.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13265/HighFidelity-Beta-Interface-PR14047-471cda8.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13265/HighFidelity-Beta-PR14047-471cda8.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13265/HighFidelity-Beta-Interface-PR14047-471cda8.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/13378/HighFidelity-Beta-PR14047-d253bad.apk

**Pull request closed without merging**